### PR TITLE
Login admin when SULS not working

### PR DESCRIPTION
**Pull request by MTsatsarov**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1124

**Summary of the changes made**:
1. Added logic in the Account controller to check the database if response from SULS is not successful and the user is admin
2.

### CODE REVIEW

**Review by shsimeonova — DISMISSED** [review]

File: Servers/UI/OJS.Servers.Ui/Controllers/AccountController.cs [hunk 1]
```diff
@@ -83,15 +87,29 @@ public async Task<IActionResult> Login([FromBody]LoginRequestModel model)
...
+                    var isInRole = await this.userManager.IsInRoleAsync(user!, GlobalConstants.Roles.Administrator);
+                    return isInRole
+                        ? this.Ok(GlobalConstants.ErrorMessages.LoggedInThroughDatabase)
+                        : this.Unauthorized(GlobalConstants.ErrorMessages.InactiveLoginSystem);
+                }
 
-            if (!signInResult.Succeeded)
+                return this.Unauthorized(GlobalConstants.ErrorMessages.InactiveLoginSystem);
+            }
+            else
```
Comment (gogo4ds) [inline Servers/UI/OJS.Servers.Ui/Controllers/AccountController.cs:104]: You don't need `else`

File: Servers/UI/OJS.Servers.Ui/Controllers/AccountController.cs [hunk 2]
```diff
@@ -83,15 +87,29 @@ public async Task<IActionResult> Login([FromBody]LoginRequestModel model)
             }
             else if (this.webHostEnvironment.IsProduction())
             {
-                return this.Unauthorized(GlobalConstants.ErrorMessages.InactiveLoginSystem);
-            }
+                var signInResult = await this.signInManager
+                    .PasswordSignInAsync(model.UserName, model.Password, model.RememberMe, false);
```
Comment (gogo4ds) [inline Servers/UI/OJS.Servers.Ui/Controllers/AccountController.cs:91]: What if the user is not admin? We will sign the user anyways?

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/Controllers/AccountController.cs [hunk 3]
```diff
@@ -83,15 +87,29 @@ public async Task<IActionResult> Login([FromBody]LoginRequestModel model)
...
 
-            var signInResult = await this.signInManager
-                .PasswordSignInAsync(model.UserName, model.Password, model.RememberMe, false);
+                if (signInResult.Succeeded)
+                {
+                    var user = await this.userManager.Users.FirstOrDefaultAsync(u => u.UserName == model.UserName);
+                    var isInRole = await this.userManager.IsInRoleAsync(user!, GlobalConstants.Roles.Administrator);
```
Comment (gogo4ds) [inline Servers/UI/OJS.Servers.Ui/Controllers/AccountController.cs:96]: Better use First() and do not force `user!`

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/Controllers/AccountController.cs (hunk 1 above)
Comment (MTsatsarov) [reply Servers/UI/OJS.Servers.Ui/Controllers/AccountController.cs]: Fixed

File: Servers/UI/OJS.Servers.Ui/Controllers/AccountController.cs (hunk 2 above)
Comment (MTsatsarov) [reply Servers/UI/OJS.Servers.Ui/Controllers/AccountController.cs]: Fixed

File: Servers/UI/OJS.Servers.Ui/Controllers/AccountController.cs (hunk 3 above)
Comment (MTsatsarov) [reply Servers/UI/OJS.Servers.Ui/Controllers/AccountController.cs]: With the new logic, there is check if the user is not null , and the "!" is removed.

**Review by gogo4ds — APPROVED** [review]

**Pull request merged**